src/main/java/technology/semi/weaviate/client/v1/auth/provider/BaseAuth.java

```diff
@@ -0,0 +1,63 @@
+package technology.semi.weaviate.client.v1.auth.provider;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.http.impl.client.HttpClientBuilder;
+import technology.semi.weaviate.client.Config;
+import technology.semi.weaviate.client.base.Serializer;
+import technology.semi.weaviate.client.base.http.HttpClient;
+import technology.semi.weaviate.client.base.http.HttpResponse;
+import technology.semi.weaviate.client.base.http.impl.CommonsHttpClientImpl;
+
+class BaseAuth {
+  protected final String OIDC_URL = "/.well-known/openid-configuration";
+  private final Serializer serializer;
+
+  BaseAuth() {
+    this.serializer = new Serializer();
+  }
+
+  @Getter
+  @AllArgsConstructor
+  public class AuthResponse {
+    String clientId;
+    String[] scopes;
+    String configuration;
+  }
+
+  @Getter
+  private class OIDCConfig {
+    String clientId;
+    String href;
+    String[] scopes;
+  }
+
+  protected AuthResponse getIdAndTokenEndpoint(Config config) throws Exception {
+    HttpClientBuilder builder = HttpClientBuilder.create();
+    HttpClient client = new CommonsHttpClientImpl(config.getHeaders(), builder::build);
+    String url = config.getBaseURL() + OIDC_URL;
+    HttpResponse response = client.sendGetRequest(url);
+    switch (response.getStatusCode()) {
+      case 404:
+        String msg = "Auth001: The client was configured to use authentication, but weaviate is configured without authentication. Are you sure this is " +
+          "correct?";
+        log(msg);
+        return null;
+      case 200:
+        OIDCConfig oidcConfig = serializer.toResponse(response.getBody(), OIDCConfig.class);
+        HttpResponse resp = client.sendGetRequest(oidcConfig.getHref());
+        if (resp.getStatusCode() != 200) {
+          String errorMessage = String.format("OIDC configuration url %s returned status code %s", oidcConfig.getHref(), response.getStatusCode());
+          throw new Exception(errorMessage);
+        }
+        return new AuthResponse(oidcConfig.getClientId(), oidcConfig.getScopes(), resp.getBody());
+      default:
+        String errorMessage = String.format("OIDC configuration url %s returned status code %s", url, response.getStatusCode());
+        throw new Exception(errorMessage);
+    }
+  }
```

Comment:
There is also the case that the response is not json, then the status code is still 200, but it is not a valid configuration. This can happen if users are behind a proxy. You can check this [issue](https://github.com/semi-technologies/weaviate-python-client/issues/177) for more infos.

In this case, the client should emit a warning (weaviate/warnings.py::auth_cannot_parse_oidc_config) and continue without authentication. A userprovided authentication header should work though. A mock-test is here: mock_tests/test_auth.py::test_auth_header_with_catchall_proxy

Reply:
In the next step I'm parsing the OIDC configuration and if there's a problem with the returned string then the exception is being thrown

Reply:
At least the setup of that one user would need that we fall back to unauthenticated access if we cannot parse the oidc info page. Not sure how common that is, but it sounds not to crazy